[PT-D] Enable nan_to_num op for sharded tensor

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78223

Differential Revision: [D36649271](https://our.internmc.facebook.com/intern/diff/D36649271)

Commit:
[PT-D] Enable nan_to_num op for sharded tensor

[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"




[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"




[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"




[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"




[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"




[ghstack-poisoned]

fduwjj commented:
@fduwjj has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36649271).

pritamdamania87 commented on torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/matrix_ops.py:
```diff
@@ -418,3 +418,8 @@ def sharded_layer_norm(args, kwargs, pg):
 )
+
+
+_register_sharded_op_on_local_tensor(
+    torch.nan_to_num,
+)
```
Isn't this an elementwise op instead of a matrix_op?

fduwjj replied:
Right. Let me move it here.

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"


Differential Revision: [D36649271](https://our.internmc.facebook.com/intern/diff/D36649271)

[ghstack-poisoned]

Commit:
Update on "[PT-D] Enable nan_to_num op for sharded tensor"


Differential Revision: [D36649271](https://our.internmc.facebook.com/intern/diff/D36649271)

[ghstack-poisoned]

fduwjj commented:
@fduwjj has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36649271).

fduwjj commented:
@pytorchbot merge this please
